src/js/apps/patients/worklist/filters_app.js

```diff
@@ -36,13 +39,28 @@ export default App.extend({
     this.showAllFiltersButtonView();
     this.showGroupsFilterView();
   },
+  getFiltersCount() {
+    const state = this.getState();
+
+    const customFiltersState = state.omit('states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterSize = size(state.get('states'));
+    const defaultStatesSize = size(this.defaultStatesFilter.map('id'));
+
+    const statesFilterCount = statesFilterSize !== defaultStatesSize ? 1 : 0;
+
+    return customFiltersSize + statesFilterCount;
+  },
   showAllFiltersButtonView() {
     const directories = Radio.request('bootstrap', 'currentOrg:directories');
 
     if (!directories.length) return;
 
     const ownerView = this.showChildView('allFilters', new AllFiltersButtonView({
-      model: this.getState(),
+      filtersCount: this.getFiltersCount(),
```

Comment:
I think we need to remove that `!directories.length` check now above.

Also the `AllFiltersButtonView` has a `modelEvents` in it which won't be used now.

Reply:
In a previous push, the `directories.length` check and `modelEvents` were removed.